pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -410,14 +448,40 @@ func (p *tinkerbellProvider) setMachinesToPXEBoot(ctx context.Context) error {
 	return errorutil.NewAggregate(errs)
 }
 
-func (p *tinkerbellProvider) SetupAndValidateDeleteCluster(ctx context.Context) error {
+func (p *tinkerbellProvider) SetupAndValidateDeleteCluster(ctx context.Context, cluster *types.Cluster) error {
 	// TODO: validations?
 	if err := setupEnvVars(p.datacenterConfig); err != nil {
 		return fmt.Errorf("failed setup and validations: %v", err)
 	}
+
+	hardwares, err := p.providerKubectlClient.GetHardwareWithOwnerName(ctx, cluster.KubeconfigFile, constants.EksaSystemNamespace)
+	if err != nil {
+		return fmt.Errorf("failed setup and validations: %v", err)
+	}
+	filteredHws, err := filterHardwareForCluster(hardwares, cluster.Name)
+	if err != nil {
+		return fmt.Errorf("failed setup and validations: %v", err)
+	}
+	p.hardwares = filteredHws
+
 	return nil
 }
 
+// filterHardwareForCluster filters hardware with ownerName label that contains cluster name.
+func filterHardwareForCluster(hardwares []tinkv1alpha1.Hardware, clusterName string) ([]tinkv1alpha1.Hardware, error) {
+	var filteredHardwareList []tinkv1alpha1.Hardware
+	for _, hw := range hardwares {
+		if strings.Contains(hw.Labels["v1alpha1.tinkerbell.org/ownerName"], clusterName) {
```

Comment:
would be nice to have this label name as a const

Reply:
Its used in 2 places so it really should be a constant somewhere.